homeassistant/components/kmtronic/switch.py

```diff
@@ -52,6 +54,19 @@ def is_on(self):
             return not self._relay.is_energised
         return self._relay.is_energised
 
+    @property
+    def device_info(self):
+        """Return device information."""
+        hub_host = self.hass.data[DOMAIN][self._config_entry_id][DATA_HUB].host
+        hostname = urllib.parse.urlsplit(hub_host).hostname
+
+        return {
+            "identifiers": {("hub", hub_host)},
```

Comment:
Is this an ip address? We don't allow that as identifier. Identifiers need to be both unique and stable.

Reply:
These devices are very basic, they don't provide any information that could be used to uniquely identify the device besides the IP.

They are configured with a fixed IP address (can't even access MAC through an API) and they literally don't have Serial Number (not even a number printed in the box)

Reply:
The last resort is the config entry entry_id.

Reply:
Thanks!